ui/tract-renderer/src/main/java/org/cru/godtools/tract/widget/PageContentLayout.java

```diff
@@ -406,6 +430,65 @@ protected LayoutParams generateLayoutParams(final ViewGroup.LayoutParams p) {
         return generateDefaultLayoutParams();
     }
 
+    //region Card Bounce Animation
+
+    /**
+     * This method will either create a new bounce Animation or start the one that is already created.
+     */
+    @UiThread
+    private void animateFirstCardView() {
+        if (mActiveCard != null) {
+            return;
+        }
+
+        for (int i = 0; i < getChildCount(); i++) {
+            final View child = getChildAt(i);
+            final LayoutParams lp = (LayoutParams) child.getLayoutParams();
+            switch (lp.childType) {
+                case CHILD_TYPE_CARD:
+                    bounceCardAnimation(child);
+                    return; // Stop after first Card View
+            }
+        }
+        stopBounceAnimation();
+
+    }
+
+    public void setAnimateCard(boolean animate) {
+        this.mBounceFirstCard = animate;
+        if (mBounceFirstCard) {
+            mHandler.sendEmptyMessageDelayed(BOUNCE_MESSAGE, DELAY_FIRST_BOUNCE_ANIMATION);
+        }
+    }
+
+    /**
+     * This method will stop the bounce Animation by setting the AnimateCard variable to false.
+     */
+    private void stopBounceAnimation() {
+        mBounceFirstCard = false;
+    }
+
+    public boolean isBounceFeatureDiscovered() {
+        return (mSettings.isFeatureDiscovered(Settings.FEATURE_TRACT_CARD_CLICKED) ||
+                mSettings.isFeatureDiscovered(Settings.FEATURE_TRACT_CARD_SWIPED));
+    }
+
+     /**
+     * This method will create a new instance of the bounce animation.
+     * @param targetView the Card View
+     */
+    @UiThread
+    private void bounceCardAnimation(final View targetView) {
+        int bounceHeight = getResources().getDimensionPixelSize(R.dimen.card_bounce_height);
+        mAnimation = ObjectAnimator.ofFloat(targetView, View.Y, targetView.getY() - bounceHeight);
+        mAnimation.setInterpolator(mBounceInterpolator);
+        mAnimation.setStartDelay(BOUNCE_ANIMATION_START_DELAY);
+        mAnimation.setDuration(mBounceInterpolator.getTotalDuration(BOUNCE_ANIMATION_DURATION_FIRST_BOUNCE));
+        mAnimation.addListener(mAnimationListener);
```

Comment:
I think we will need a separate animation listener for bounce animations that doesn't dispatch the active card change event.